patches/nginx-1.17.8-set_uri_security_issue.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/src/http/modules/ngx_http_static_module.c b/src/http/modules/ngx_http_static_module.c
+index 282d6ee..cb5b6c1 100644
+--- a/src/http/modules/ngx_http_static_module.c
++++ b/src/http/modules/ngx_http_static_module.c
+@@ -44,6 +44,20 @@ ngx_module_t  ngx_http_static_module = {
+     NGX_MODULE_V1_PADDING
+ };
+ 
++static ngx_int_t
++ngx_http_check_safe_location(u_char *str, size_t len)
++{
++    u_char ch;
++    size_t i;
++
++    for (i = 0; i < len; i++) {
```

Comment:
Maybe it's more efficient to use pointers directly as in

```
char *last = uri + len;

for (p = uri; p != last; p++) {
    c = *p;
```

Also note the naming. `str` should be `uri` to be more meaningful and `ch` is better to be `c`.